### ISSUE TITLE
Validation stage

### ISSUE THREAD

**Issue by Jamesflynn1** [Jamesflynn1/FiniteMarketModel#2]
Select best hyperparameter on the validation set (use Wasserstein distance), for each stock and use this for evaluation stage.
